spec/controllers/iiif_controller_spec.rb

```diff
@@ -165,6 +186,47 @@ def stanford_restricted?
           expect(logout_service['label']).to eq 'Logout'
         end
       end
+
+      context 'when the image is location-restricted' do
+        before do
+          allow(stub_metadata_object).to receive(:restricted_by_location?).and_return(true)
+          allow(stub_metadata_object).to receive(:restricted_locations).and_return([:spec])
+        end
+
+        context 'and the user is not in that location' do
+          it 'advertises an authentication service' do
+            expect(image_info['service']).to be_present
+            expect(image_info['service']['profile']).to eq 'http://iiif.io/api/auth/1/external'
+          end
```

Comment:
might be nice to also check `'label'` and/or `'failureHeader'` for the expected text?  though i don't think that's super-important, and wouldn't object if you wanted to leave it out.   would also be nice to test `'failureDescription'` to see that it includes the expected location, though the method that gets the location list at least seems to be tested in `stacks_rights_spec.rb`.